### PR TITLE
Add mysql_index module

### PR DESCRIPTION
**Pull request by caphrim007**
This module provides the ability for ansible to add and remove indices
from a table in MySQL without needing to use other, more raw, modules to
accomplish similar goals.

### CODE REVIEW

File: library/database/mysql_index [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: mysql_index
+short_description: Adds or removes an index from a MySQL table.
+description:
+   - Adds or removes an index from a MySQL table.
+version_added: "1.6"
```
Comment (omgjlk) [inline library/database/mysql_index:28]: This will probably have to be 1.7, as 1.6 is already released.

**Comment by mpdehaan** [comment]
Hi, 

While this is interesting I think most people will apply this through a database schema upgrade or execute a SQL script.

You are welcome to continue to use this module though we are unlikely to include it in Ansible core.

Feel free to make a post about it to ansible-project list to let people know about it.

Thanks!

**Comment by caphrim007** [comment]
@mpdehaan hey michael, actually, we were using a SQL script before we developed this module. We created it because re-using the SQL script caused duplicate indices to be created (non idempotent). We also were able to wrap this module in a with_items that resulted in a cleaner-looking playbook and elimination of said SQL script altogether.

**Comment by srgvg** [comment]
FYI, @caphrim007 if you are interested, I'm collecting various plugins and modules not in core here: https://github.com/ginsys/ansible-plugins

**Pull request closed without merging**